### PR TITLE
Refactor cancellation to not use stately collections

### PR DESCRIPTION
**Pull request by benasher44**
I want to get us off of stately. With atomicfu now supporting multiplatform, stately will become less relevant for CoroutineWorker, so I want to see if we can reduce CoroutineWorker's dependencies. This refactors to use a local implementation of AtomicBoolean for cancellation.

### CODE REVIEW

**Comment by benasher44** [comment]
There will be more PRs coming to get us off of stately. This is just the first one.

File: build.gradle.kts [hunk 1]
```diff
@@ -55,8 +54,8 @@ kotlin {
         }
         val jvmTest by getting {
             dependencies {
-                implementation("org.jetbrains.kotlin:kotlin-test:$kotlinVersion")
-                implementation("org.jetbrains.kotlin:kotlin-test-junit:$kotlinVersion")
+                implementation("org.jetbrains.kotlin:kotlin-test")
```
Comment (benasher44) [inline build.gradle.kts]: I learned from gradle folks at Kotlinconf that we don't need to specify kotlin version for these. It'll just the right version for the version of Kotlin you picked.

**Comment by kpgalligan** [comment]
Aw, stately. Just FYI, collections will probably get a refactor, but the Atomics will probably stick around. On the JVM, atomic-fu is doing more "magic" and has some restrictions and concerns outside of just expect/actual. Not that you should keep stately, but still.

**Comment by benasher44** [comment]
Haha

**Comment by benasher44** [comment]
Haha still love/use stately at Autodesk! Just realized now that my understanding of atomics in K/N has grown since the initial write here, and I can pare down our use of 3rd-party deps here :)

**Comment by kpgalligan** [comment]
Stately's expect/actuals are not complex. I definitely think libraries should do their own.

**Comment by benasher44** [comment]
Agree! Should be able to get there for CoroutineWorker 🙏

File: src/nativeMain/kotlin/com/autodesk/coroutineworker/CoroutineWorker.kt [hunk 2]
```diff
@@ -150,3 +110,55 @@ actual class CoroutineWorker {
...
+     */
+    var completed: Boolean
+        get() = isSet(completedBit)
+        set(value) = updateValue(completedBit, value)
+
+    /**
+     * Updates the value with the bit, setting or un-setting it
+     */
+    private fun updateValue(bit: Int, set: Boolean) {
+        value.value = if (set) {
```
Comment (arkivanov) [inline src/nativeMain/kotlin/com/autodesk/coroutineworker/CoroutineWorker.kt]: Nut sure but if this method is called from different threads then there is a race condition here. You first read the current value then change and then save. This operation should be atomic.

Comment (benasher44) [reply src/nativeMain/kotlin/com/autodesk/coroutineworker/CoroutineWorker.kt]: Great catch! Will fix :)

Comment (arkivanov) [reply src/nativeMain/kotlin/com/autodesk/coroutineworker/CoroutineWorker.kt]: You need something like [this](https://github.com/badoo/Reaktive/blob/93a34ebe23ad3cf3584def7fb30247605ec01fd3/utils/src/commonMain/kotlin/com/badoo/reaktive/utils/atomic/AtomicReferenceExt.kt#L24) but for `AtomicInt`

**Pull request merged**